spec/cart_spec.rb

```diff
@@ -1,16 +1,40 @@
+# frozen_string_literal: true
+
 require 'spec_helper'
 require_relative '../cart'
+require_relative '../product'
 
 RSpec.describe Cart do
-  describe '#add_product' do
-    subject { described_class.new }
+  let(:product) { Product.new(3, 'Ror', 300.20) }
+  let(:product2) { Product.new(1, 'Adv', 20.22) }
+  let(:cart) { Cart.new }
 
-    it 'adds product to items list' do
-      subject.add_product(1)
-      expect(subject.products).to eq([1])
+  describe '#add_product_to_cart' do
+    it 'add' do
+      cart.add_product_to_cart(product)
+      expect(cart.products_in_cart).to include(product.id)
     end
   end
-end
 
+  describe '#remove_product_from_cart' do
+    it 'remove' do
+      cart.add_product_to_cart(product)
```

Comment:
Podchodząc purytańsko to testuje i dodawanie i usuwanie, jak ci się wywali ten test, to nie wiesz która metoda nie działa. Żeby tego uniknąć Cart powinien być zainicjalizowany z produktem